rclcpp/include/rclcpp/timer.hpp

```diff
@@ -317,6 +339,7 @@ class GenericTimer : public TimerBase
   RCLCPP_DISABLE_COPY(GenericTimer)
 
   FunctorT callback_;
+  rcl_timer_call_info_t timer_call_info_;
```

Comment:
@atsogias I don't like this, this introduces a potential race condition. I'll push a fix for this.